### PR TITLE
feat: add checkbox test

### PR DESCRIPTION
**Pull request by Samantha-KY**
# Submit a pull request

- [x] This is not a duplicate of an existing pull request.
- [x] No existing features have been broken without good reason.
- [x] Your commit messages are detailed
- [x] The code style guideline have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [x] Tests have been added or updated to reflect your changes.
- [x] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

This PR add test for the checkbox component

:question:

## Breaking Changes

Describe what features are broken by this pull request and why, if any.

:question:

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

1. :question:

## Other Relevant Information

Provide any other important details below.

### CODE REVIEW

File: src/components/ui/Checkbox.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(function ({ disable
     "cursor-pointer": !disabled,
   });
 
-  return <input ref={ref} {...props} type="checkbox" disabled={disabled} className={checkboxClassName} style={styles} />;
+  return <input data-testid="checboxId" ref={ref} {...props} type="checkbox" disabled={disabled} className={checkboxClassName} style={styles} />;
```
Comment (serapieTuyishime) [inline src/components/ui/Checkbox.tsx:40]:
```suggestion
  return <input data-testid="checkboxId" ref={ref} {...props} type="checkbox" disabled={disabled} className={checkboxClassName} style={styles} />;
```

File: __tests__/components/ui/Checkbox.test.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        )
+        const checkbox = screen.getByTestId("checboxId"); 
+        expect(checkbox).toBeInTheDocument()
+    })
+
+    it("should render the checkbox with community style", () => {
+        render(<Checkbox communityStyles/>)
+        const checkboxStyle = screen.getByTestId("checboxId"); 
+        expect(checkboxStyle).toBeInTheDocument()
+        expect(checkboxStyle).toHaveStyle({color: "#0000FF"})
```
Comment (Jonath-z) [inline __tests__/components/ui/Checkbox.test.tsx:18]: We need to avoid testing the exact community color, we can only test of the color is not undefined or empty string. This is because the community color can changes at anytime which in this case will break the test.

File: __tests__/components/ui/Checkbox.test.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    })
+
+    it("should render the checkbox as disabled", () => {
+        render(<Checkbox disabled />);
+        const checkbox = screen.getByTestId("checboxId"); 
+        expect(checkbox).toBeInTheDocument();
```
Comment (Jonath-z) [inline __tests__/components/ui/Checkbox.test.tsx:24]: This already tested, we dont need to check it's in the document again

File: __tests__/components/ui/Checkbox.test.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        expect(checkbox).toBeDisabled();
+    });
+
+    it("should render the checkbox checked", () => {
+        render(<Checkbox checked />);
+        const checkbox = screen.getByTestId("checboxId"); 
+        expect(checkbox).toBeInTheDocument();
```
Comment (Jonath-z) [inline __tests__/components/ui/Checkbox.test.tsx:31]: here as well

File: __tests__/components/ui/Checkbox.test.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    it("should call the onChange function when checkbox is clicked", () => {
+        const onChange = jest.fn();
+        render(<Checkbox onChange={onChange} />);
+        const checkbox = screen.getByTestId("checboxId"); 
+        expect(checkbox).toBeInTheDocument();
+        expect(onChange).not.toHaveBeenCalled();
+        fireEvent.click(checkbox);
+        expect(onChange).toHaveBeenCalled();
+    });
```
Comment (Jonath-z) [inline __tests__/components/ui/Checkbox.test.tsx:43]: This is nice already, Also check when the click fires the checkbox is checked and when clicked again is not checked.

**Review by Jonath-z — CHANGES_REQUESTED** [review]

**Review by Jonath-z — APPROVED** [review]

**Review by serapieTuyishime — APPROVED** [review]
